modules/decrypt-node/test/parse_header_stream.test.ts

```diff
@@ -39,15 +42,50 @@ describe('ParseHeaderStream', () => {
       await testStream(cmm, data.slice(0, i))
     }
   })
+
+  it('Precondition: If maxHeaderSize was set I can not buffer a header larger than maxHeaderSize.', async () => {
+    const completeHeaderLength = 73
+    const data = Buffer.from(
+      fixtures.base64CiphertextAlgAes256GcmIv12Tag16HkdfWith4Frames(),
+      'base64'
+    )
+    const cmm = new NodeDefaultCryptographicMaterialsManager(
+      fixtures.decryptKeyring()
+    )
+
+    /* Starting from 1
+     * test every maxHeaderSize up to
+     * 1 less than the actual header size.
+     * There is a subtle JS assumption here,
+     * that 0 is false and thus an irrational value
+     * to push into `maxHeaderSize`.
+     */
+    for (let i = 1; completeHeaderLength > i; i++) {
+      await expect(testStream(cmm, data, { maxHeaderSize: i })).rejectedWith(
+        Error,
+        'maxHeaderSize exceeded.'
+      )
+    }
+
+    /* Picking up from the exact header size
+     * test every maxHeaderSize equal to
+     * or greater than the actual header size
+     * (up to the message size).
+     */
+    for (let i = completeHeaderLength; data.byteLength > i; i++) {
+      await testStream(cmm, data.slice(0, i), { maxHeaderSize: i })
+    }
```

Comment:
It's not immediately obvious to me that `data.byteLength > completeHeaderLength` ever holds (even if it currently does), and if it doesn't hold then nothing is tested here. Would it make sense to assert that `data.byteLength > completeHeaderLength`, i.e. that the loop runs at least once?